src/Symfony/Component/Uid/UuidFactory.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Uid;
+
+class UuidFactory
+{
+    private $defaultNamedVersion;
+    private $defaultTimedVersion;
+
+    /**
+     * @var Uuid|null
+     */
+    private $defaultNamespace;
+
+    /**
+     * @var Uuid|null
+     */
+    private $defaultNode;
+
+    /**
+     * @param string|null $defaultNamedVersion UuidV5 or UuidV3 FQCN
+     * @param string|null $defaultTimedVersion UuidV6 or UuidV1 FQCN
+     */
+    public function __construct(string $defaultNamedVersion = null, string $defaultTimedVersion = null)
+    {
+        $this->defaultNamedVersion = $defaultNamedVersion ?? UuidV5::class;
+        $this->defaultTimedVersion = $defaultTimedVersion ?? UuidV6::class;
+    }
+
+    /**
+     * @param string|null $version UuidV5 or UuidV3 FQCN
+     *
+     * @return UuidV5|UuidV3
+     */
+    public function createNamed(string $name, Uuid $namespace = null, string $version = null): Uuid
+    {
+        switch ($version ?? $this->defaultNamedVersion) {
+            case UuidV5::class:
+                return Uuid::v5($namespace ?? $this->defaultNamespace ?? $this->createRandom(), $name);
+            case UuidV3::class:
+                return Uuid::v3($namespace ?? $this->defaultNamespace ?? $this->createRandom(), $name);
+        }
+
+        throw new \InvalidArgumentException(sprintf('The "%s" version is unsupported.', $version));
+    }
+
+    /**
+     * @param string|null $timestamp Count of 100-nanosecond intervals since the UUID epoch 1582-10-15 00:00:00
```

Comment:
this is going to be a pain to compute
how do you think one will be able to generate/get such timestamps?
should we provide a method to extract them from existing UUIDs?

Reply:
For the record we decided to use the same format than `getTime()` (seconds since unix epoch) for simplicity. If someone wants to use ulid and uuid "real" timestamp format, they'll have to do the conversion.